src/game/client/gui/controlbar/controlbarcommand.cpp

```diff
@@ -125,9 +134,882 @@ const Image *ControlBar::Calculate_Veterancy_Overlay_For_Thing(const ThingTempla
+                                    m_commandWindows[i]->Win_Hide(true);
+                                }
+
+                                Set_Control_Command(m_commandWindows[i], command_button);
+                            }
+                        }
+                    }
+                }
+
+                if (last_exit_index >= 0) {
+                    PopulateInvButtonData data;
+                    data.controls = m_commandWindows;
+                    data.curr_index = first_exit_index;
+                    data.max_index = last_exit_index;
+                    data.transport = transport;
+                    contain->Iterate_Contained(Populate_Inv_Data_Callback, &data, false);
+                }
+
+                m_lastRecordedInventoryCount = contain->Get_Contain_Count();
+            }
+        }
+    }
+}
+
+void ControlBar::Populate_Command(Object *obj)
+{
+    Player *player = obj->Get_Controlling_Player();
+    Reset_Contain_Data();
+    Reset_Build_Queue_Data();
+    const CommandSet *command_set = g_theControlBar->Find_Command_Set(obj->Get_Command_Set_String());
+
+    if (command_set == nullptr) {
+        for (int i = 0; i < CommandSet::MAX_COMMAND_BUTTONS; i++) {
+            if (m_commandWindows[i] != nullptr) {
+                m_commandWindows[i]->Win_Hide(true);
+            }
+        }
+
+        return;
+    }
+
+    if (obj->Get_Contain() != nullptr) {
+        ContainModuleInterface *contain = obj->Get_Contain();
+
+        if (contain->Is_Displayed_On_Control_Bar()) {
+            Do_Transport_Inventory_UI(obj, command_set);
+        }
+    }
+
+    for (int i = 0; i < CommandSet::MAX_COMMAND_BUTTONS; i++) {
+        if (m_commandWindows[i] == nullptr) {
+            continue;
+        }
+
+        const CommandButton *command_button = command_set->Get_Command_Button(i);
+
+        if (command_button == nullptr || (command_button->Get_Options() & COMMAND_OPTION_SCRIPT_ONLY) != 0) {
+            m_commandWindows[i]->Win_Hide(true);
+            continue;
+        }
+
+        if (command_button->Get_Command() != GUI_COMMAND_EXIT_CONTAINER) {
+            m_commandWindows[i]->Win_Hide(false);
+            m_commandWindows[i]->Win_Enable(true);
+            Set_Control_Command(m_commandWindows[i], command_button);
+
+            if ((command_button->Get_Options() & COMMAND_OPTION_NEED_SPECIAL_POWER_SCIENCE) != 0) {
+                const SpecialPowerTemplate *power = command_button->Get_Special_Power();
+
+                if (power != nullptr) {
+                    if (power->Get_Required_Science() != SCIENCE_INVALID
+                        && command_button->Get_Command() != GUI_COMMAND_PURCHASE_SCIENCE
+                        && command_button->Get_Command() != GUI_COMMAND_PLAYER_UPGRADE
+                        && command_button->Get_Command() != GUI_COMMAND_OBJECT_UPGRADE) {
+
+                        if (!player->Has_Science(power->Get_Required_Science())) {
```

Comment:
Potentially referencing NULL. See 10 lines below, it checks player for NULL.

Reply:
Original code does this.